tests/test_api.py

```diff
@@ -200,6 +205,25 @@ def test_sign_verify(self):
         with self.assertRaises(tuf.exceptions.UnsignedMetadataError):
             targets_key.verify_signature(metadata_obj)
 
+        # Test failure on broken public key data (securesystemslib CryptoError)
+        public = timestamp_key.keyval["public"]
+        timestamp_key.keyval["public"] = "ffff"
+        with self.assertRaises(tuf.exceptions.UnsignedMetadataError):
+            timestamp_key.verify_signature(metadata_obj)
+        timestamp_key.keyval["public"] = public
+
+        # Test failure with invalid signature (securesystemslib FormatError)
+        sig = metadata_obj.signatures[timestamp_keyid]
+        correct_sig = sig.signature
+        sig.signature = "foo"
+        with self.assertRaises(tuf.exceptions.UnsignedMetadataError):
+            timestamp_key.verify_signature(metadata_obj)
+
+        # Test failure with valid but incorrect signature
```

Comment:
Which exception does this test case test?
Also, I am not sure I see a test case when `sslib_keys.verify_signature` will call `UnsupportedAlgorithmError`.
Is this a test case for that?

Reply:
Signature verification failure is not an exception in Securesystemslib.

UnsupportedAlgorithm question is relevant, I wonder if there was a reason I didn't include it: I'll have to check.

Reply:
I did not see a reason not to so I added the test... we could still test many more possible cases but AFAIK they will all lead to exceptions that we now handle (and test) so I think this might be good enough